daintree/run_fit_models.py

```diff
@@ -170,25 +170,59 @@ def gather_ensemble_tasks(
         cors = (
             pd.DataFrame(cors)
             .reset_index()
-            .rename(columns={"index": "gene", 0: "pearson"})
+            .rename(columns={"index": "target_variable", 0: "pearson"})
         )
         cors["model"] = model
 
         all_cors.append(cors)
 
     all_cors = pd.concat(all_cors, ignore_index=True)
-    ensemble = all_features.merge(all_cors, on=["gene", "model"])
+    ensemble = all_features.merge(all_cors, on=["target_variable", "model"])
+    
+    ### The following code is implemented this way due to a
+    # PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.
+    # De-fragment the DataFrame
+    ensemble = ensemble.copy()
 
-    # Get the highest correlation across models per "gene" (entity)
-    ensemble["best"] = ensemble.groupby("gene")["pearson"].rank(ascending=False) == 1
+    # Get the highest correlation across models per "target_variable" (entity)
+    ranked_pearson = ensemble.groupby("target_variable")["pearson"].rank(ascending=False)
+    ensemble["best"] = (ranked_pearson == 1)
 
-    ensb_cols = ["gene", "model", "pearson", "best"]
+    ensb_cols = ["target_variable", "model", "pearson", "best"]
 
-    for i in range(top_n):
-        cols = ["feature" + str(i), "feature" + str(i) + "_importance"]
-        ensb_cols += cols
+    # Iterate over each row in the ensemble
+    for index, row in ensemble.iterrows():
+        # Get the target variable name from the ensemble
+        target_variable = row['target_variable']
+        
+        # Extract the corresponding target data from `targets` DataFrame
+        y = targets[target_variable]
+
+        # Iterate over feature columns in the ensemble
+        for i in range(top_n):  # Assuming there are 50 features (feature0 to feature49)
+            feature_col = f'feature{i}'
+            feature_name = row[feature_col]  # Get the feature name listed in the ensemble row
+            
+            if feature_name in features.columns:
```

Comment:
If `feature_name` isn't in `features` that sounds like something has gone wrong.  How do we get an important feature which doesn't exist?

 I'd make this into an assertion rather than silently leaving a NA behind.